src/material/chips/chip-input.ts

```diff
@@ -105,21 +121,50 @@ export class MatChipInput implements MatChipTextControl, OnChanges {
     this._inputElement = this._elementRef.nativeElement as HTMLInputElement;
   }
 
-  ngOnChanges() {
+  ngOnChanges(): void {
     this._chipList.stateChanges.next();
   }
 
+  ngOnDestroy(): void {
+    this.chipEnd.complete();
+  }
+
+  ngAfterContentInit(): void {
+    this._focusLastChipOnBackspace = this.empty;
+  }
+
   /** Utility method to make host definition/tests more clear. */
   _keydown(event?: KeyboardEvent) {
-    // Allow the user's focus to escape when they're tabbing forward. Note that we don't
-    // want to do this when going backwards, because focus should go back to the first chip.
-    if (event && event.keyCode === TAB && !hasModifierKey(event, 'shiftKey')) {
-      this._chipList._allowFocusEscape();
+    if (event) {
+      // Allow the user's focus to escape when they're tabbing forward. Note that we don't
+      // want to do this when going backwards, because focus should go back to the first chip.
+      if (event.keyCode === TAB && !hasModifierKey(event, 'shiftKey')) {
+        this._chipList._allowFocusEscape();
+      }
+
+      if (event.keyCode === BACKSPACE && this._focusLastChipOnBackspace) {
```

Comment:
comment what this block is for

Reply:
Added a comment, if you have a suggestion for how to make it shorter/clearer, I'll appreciate it 😅

Reply:
What you have sounds good to me, minor nit: "And" on the 3rd line doesn't need to be capitalized

Reply:
Will change... I have this habit for a couple of years now when writing docs, I don't quite know why 🤦🏻‍♂️